lms/services/user.py

```diff
@@ -139,6 +142,106 @@ def _user_search_query(self, application_instance_id, user_id) -> Select:
 
         return query
 
+    def get_users_for_assignment(
+        self,
+        role_scope: RoleScope,
+        role_type: RoleType,
+        assignment_id: int,
+        h_userids: list[str] | None = None,
+    ):
+        """Get the users that belong to one assignment."""
+        query = (
+            select(LMSUser)
+            .distinct()
+            .join(
+                LMSUserAssignmentMembership,
+                LMSUserAssignmentMembership.lms_user_id == LMSUser.id,
+            )
+            .where(
+                LMSUserAssignmentMembership.assignment_id == assignment_id,
+                LMSUserAssignmentMembership.lti_role_id.in_(
+                    select(LTIRole.id).where(
+                        LTIRole.scope == role_scope, LTIRole.type == role_type
+                    )
+                ),
+            )
+        )
+        if h_userids:
+            query = query.where(LMSUser.h_userid.in_(h_userids))
+
+        return query.order_by(LMSUser.display_name, LMSUser.id)
+
+    def get_users_for_course(
+        self,
+        role_scope: RoleScope,
+        role_type: RoleType,
+        course_id: int,
+        h_userids: list[str] | None = None,
+    ):
+        """Get the users that belong to one course."""
+        query = (
+            select(LMSUser)
+            .distinct()
+            .join(
+                LMSCourseMembership,
+                LMSCourseMembership.lms_user_id == LMSUser.id,
+            )
+            .join(LMSCourse, LMSCourse.id == LMSCourseMembership.lms_course_id)
+            # course_id is the PK on Grouping, we need to join with LMSCourse by authority_provided_id
+            .join(
+                Grouping,
```

Comment:
To avoid this will have to base the API on authority_provided_ids.

We could still have thet API based on course_id, translate them to "lms_course_id" before calling the service and tackle the API level later.

IMO this PR is already big enough.